lib/irb/completion.rb

```diff
@@ -86,6 +86,14 @@ def retrieve_files_to_require_from_load_path
         )
     end
 
+    def command_completions(preposing, target)
+      if preposing.empty? && !target.empty?
+        IRB::ExtendCommandBundle.command_names.select { _1.start_with?(target) }
```

Comment:
Maybe `IRB::ExtendCommandBundle.command_names` can be (lazily) cached somewhere so we don't compute it repeatedly?

Reply:
I think it's already cached
```ruby
def self.command_names
  command_override_policies.keys.map(&:to_s)
end
```
`command_override_policies` is cached, `.keys.map(&:to_s)` is not a heavy instruction compared to `select{_1.start_with?(target)}`, so the effect of caching names is small, I think.